### ISSUE TITLE
node_modules folder not being created per function

### ISSUE THREAD

**Issue by hicksy** [arc-repos/arc-workflows#41]
I started another project using workflows `2.1.0` - there appears to be an issue during deployment - if I view the lamda function within AWS Console it only contains the `index.js` & `package.json` files. 

There's no `node_modules` folder deployed, and therefore no `@architect/functions` to import. 

To test this theory I've manually run `npm install @architect/functions` within my lamda folder and re-deployed. This works, without error, and I get the expected response when hitting the endpoint. 

Guessing it's something at the point of creating the deployable package??

Node: 8.2.1
Npm: 5.4.1 

This is the first project I've created using `2.1.0`, have several `1.x.x` projects created / deployed successfully from this machine.  👍

**Comment by brianleroux** [comment]
Cool thx for the report -- just attempting to reproduce rn

**Comment by brianleroux** [comment]
hmmm, I was unable to reproduce @hicksy ---might be the Node version or maybe a hiccup w the network when you generated the source. Could you try upgrading you Node and see if it is "fixed"?

```
#! node -v
v8.9.2
#! npm -v
5.5.1
```

**Comment by hicksy** [comment]
Thanks for looking into it @brianleroux - I've done a little more testing, appears to only affect Windows 10 (possibly just my Win10 setup)

The original issue was present when I deployed from Win 10 and OS X, but crucially the project was created on Win 10 originally. 

I've created a couple of test endpoints from each OS, using the same basic .arc with just a JSON /status endpoint

OS X 🆗   
(https://jmr81uyidd.execute-api.eu-west-2.amazonaws.com/staging/status)
```
osx-test ➜ node -v
v8.9.1
osx-test ➜ npm -v
5.5.1
osx-test ➜
```

Win 10 (bash / WSL) 🆗 
(https://vxw5nny355.execute-api.eu-west-2.amazonaws.com/staging/status)
```
# node -v
v8.2.1
# npm -v
5.4.1
```

Win 10 - creates without error, but no `node_modules`  
(https://37tpbtzirc.execute-api.eu-west-2.amazonaws.com/staging/status)
```
PS> node -v
v8.9.1
PS> npm -v
5.6.0
PS>
```

I guess this is likely pointing to an issue with my Win 10 setup - I'll remove / install node when I get a sec and see where that gets me, and let you know.

**Comment by brianleroux** [comment]
🆒 Windows support is important to us so pls do keep us updated. I'm planning to pick up a Surface when I get back to N America in the next week or so. 😊

**Comment by hicksy** [comment]
Cool!

Managed to grab some time to investigate a little more. Removed, then re-installed Node, and created the same basic json /status endpoint - same issue I'm afraid, no `node_modules` 

```
node -v
v8.9.3

npm -v
5.5.1
```

Did some digging around using the Debugger and found where I think the problem lies. The following code style doesn't seem to be compatible in a Windows environment:

```Node
exec(`
    cd ${localPath} && \
    npm i @architect/functions --save --production
`,
function _exec(err) {
    if (err) {
    console.log(err)
    }
    callback()
})
```

Instead for Windows compatibility it needs to be a single line command:

```Node
exec('npm i @architect/functions --save --production ',{
    cwd: localPath
},
function _exec(err) {
    if (err) {
    console.log(err)
    }
    callback()
})
```
Apparently Windows treats each new line as a separate command (ref: https://stackoverflow.com/a/44171391)

I made this change to `@architect/workflows/src/create/aws/_create-code.js` and the `node_modules` are now correctly installed within the lambda 👍 

But, this only works when serving the local arc-sandbox server..... 😢 Guess my first FOSS pull request wont come as soon as I'd hoped!

I did have a little look into what might be going on with this second part of the problem, but mainly drew blanks - thought it might be due to `cipm` and something odd with `package-lock.json`, and while it may be I guess, I commented that block out from the deploy steps and it didn't help. 

One thing that might aid diagnosis - I manually zipped the content of the local lambda function and uploaded to AWS - it worked perfectly (that was with cipm re-enabled btw)

Hope this helps
